torch/_inductor/lowering.py

```diff
@@ -5715,8 +5746,12 @@ def reduce_min(x, dim=None, keepdim=False):
 
 
 register_lowering(prims.xor_sum)(make_reduction("xor_sum"))
-reduce_amax = register_lowering(aten.amax)(make_reduction("max"))
-reduce_amin = register_lowering(aten.amin)(make_reduction("min"))
+reduce_amax = register_lowering(aten.amax, type_promotion_kind=None)(
+    make_reduction("max")
+)
+reduce_amin = register_lowering(aten.amin, type_promotion_kind=None)(
+    make_reduction("min")
+)
```

Comment:
Something about the set of changes made in `register_lowering` exposed a bug here, where passing `bool` types to `amax` or `amin` reductions would promote the output to integers. This changes fixes what would be a failing test.

Reply:
Hmm, so this is going to have the effect of making non-tensor kwargs participate in type promotion. I'm not sure if this is going to have any unintended consequences. As the comment says:
```
# FIXME that's a crude approximation for promoting args
```

Our type promotion logic could certainly be better. One thing we could do is just take a peak at the dtype of the FakeTensor output of `V.graph.current_node` (not for this pr).

In this PR, could we ignore the non-Tensor kwargs for type promotion ? cc @ezyang, @bdhirsh thoughts

Reply:
@eellison It should be relatively easy to make non-tensors not participate in type promotion, if we want to go that way.

Reply:
Yea, do you mind if we do that ? We could make them participate in a follow up. I think that make this less stack less risky to land. As I think I said, opinfos are great but I don't honestly know to what extent they exercise all of the type promotion combinations especially as it pertains to kwargs.

Reply:
@eellison Tested locally, and seems to be working. Pushed.